wasm_runtime/src/config.rs

```diff
@@ -210,7 +210,42 @@ impl WasmConfig {
         wasm_config.wasi_mapdirs.push((wasi_map.to_string(), wasi_dir.to_string()));
         Ok(())
     }
+    // Returns a version of path with all slashes converted to forward
+    fn path_to_unix_slashes(path: &Path) -> String {
+        path.to_string_lossy().into_owned().replace("\\", "/")
+    }
+    fn find_longest_map(mapdirs: &[(String, String)], path: &str) -> Option<String> {
+        let cleaned_path = clean(Self::path_to_unix_slashes(Path::new(path)));
+        let mut ordered = mapdirs.to_vec();
+        ordered.sort_by_key(|(_, from)| Path::new(from).components().count());
```

Comment:
Sounds hacky, but it's also possible to count `std::path::MAIN_SEPARATOR` to identify longitude; this would not involve allocating.

Reply:
That's a really good alternative 👏

Reply:
Or even easier, we can sort by whole path string longitude, and pick the first one that is a su path of the needle

Reply:
In this case, I think the goal is to calculate the "deepest" path, but not the longest. So for example, `/a/b/c/d` should be selected instead of `/a/superlongfoldername`.

Reply:
Yes, string length is not what we want. I used components to make sure I counted the right thing (it does some work to remove .. and so on) but as we are working with the cleaned path, we can assume it is safe to just count slashes. In any case, we should count "/", not std::path::MAIN_SEPARATOR, right? As std::path::MAIN_SEPARATOR will depend on the platform and on Windows it will be "\"

Reply:
Yes, you are right, we have to count the separators as written in the configuration, not std::path::MAIN_SEPARATOR.